File: Data Structures & Algorithms/partition-equal-subset-sum/submission-0.go
func canPartition(nums []int) bool {
	total := 0 
	for _, v := range nums {
		total += v
	}

	if total%2 != 0 {
		return false
	}

	target := total / 2
	dp := make([]bool, target + 1)
	dp[0] = true
	for _, num := range nums {
		for c := target; c >= num; c-- {
			dp[c] = dp[c] || dp[c-num]
		}
	}

	return dp[target]
}

File: Data Structures & Algorithms/target-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    findTargetSumWays(nums, target) {
        // backtrack
        const backtrack = (i, total) => {
            if(i === nums.length) {
                // 1 is correct way
                return total === target ? 1 : 0
            } 
            return (
                // plus branch
                backtrack(i + 1, total + nums[i]) + 
                // sub branch
                backtrack(i + 1, total - nums[i])
            )
        }

        return backtrack(0, 0)
    }
}
